### PR TITLE
test: improve coverage for some files

### PR DESCRIPTION
**Pull request by saumyatalwani**
> [!IMPORTANT]
>
> 1. Make sure you have read our [contribution guidelines](https://github.com/langgenius/dify/blob/main/CONTRIBUTING.md)
> 1. Ensure there is an associated issue and you have been assigned to it
> 1. Use the correct syntax to link this PR: `Fixes #<issue number>`.

## Summary

* **Tests**
  * Dramatically expanded unit and integration tests across copy/clipboard, secret inputs, video player, QR code, chat rendering, forms, inputs, and many edge cases.

* **Bug Fixes**
  * Tooltip and placeholder fallbacks made null-safe; copy behavior handles non-string/undefined values; improved stability for video, QR, and selection interactions.

* **Refactor**
  * Added test hooks and lightweight UI markup tweaks to improve testability and consistency.

* **Accessibility**
  * Enhanced checkbox keyboard interaction, ARIA attributes, and focus behavior.
  
  
  
  fixes: #32152 

## Screenshots

| Before | After |
|--------|-------|
| ... | ... |

## Checklist

- [ ] This change requires a documentation update, included: [Dify Document](https://github.com/langgenius/dify-docs)
- [x] I understand that this PR may be closed in case there was no previous discussion or issues. (This doesn't apply to typos!)
- [x] I've added a test for each change that was introduced, and I tried as much as possible to make a single atomic change.
- [x] I've updated the documentation accordingly.
- [x] I ran `make lint` and `make type-check` (backend) and `cd web && npx lint-staged` (frontend) to appease the lint gods

### CODE REVIEW

File: web/app/components/base/checkbox/index.tsx [hunk 1]
```diff
@@ -40,10 +39,23 @@ const Checkbox = ({
...
+          if (event.key === ' ')
+            event.preventDefault()
+          onCheck?.(event)
+        }
+      }}
       data-testid={`checkbox-${id}`}
+      role="checkbox"
+      aria-checked={indeterminate ? 'mixed' : !!checked}
+      aria-disabled={!!disabled}
+      tabIndex={disabled ? -1 : 0}
```
Comment (CodingOnStar) [inline web/app/components/base/checkbox/index.tsx:55]: Either switch to a semantic checkbox control that can be labeled correctly, or extend the component API to accept aria-label / aria-labelledby and update the touched call sites to provide one before shipping the new keyboard support

**Comment by CodingOnStar** [comment]
Please remove v8 ignore from branches that can be exercised through tests or minor refactors, and keep it only for truly untestable environment-specific paths with a precise justification. Coverage gains here should come primarily from tests, not exclusions

**Review by CodingOnStar — COMMENTED** [review]
This commit adds more comments instead of removing the redundant ones. I don't believe these comments are necessary

**Comment by saumyatalwani** [comment]
> This commit adds more comments instead of removing the redundant ones. I don't believe these comments are necessary

haven't added any new comments, only added reasonable justifications for existing ones. react-i18n branches are unreachable without adding a mock (which is restricted).

**Comment by CodingOnStar** [comment]
> > This commit adds more comments instead of removing the redundant ones. I don't believe these comments are necessary
> 
> haven't added any new comments, only added reasonable justifications for existing ones. react-i18n branches are unreachable without adding a mock (which is restricted).

Sounds good :)

**Review by CodingOnStar — APPROVED** [review]
